orangecontrib/text/corpus.py

```diff
@@ -55,6 +56,8 @@ def __init__(self, X=None, Y=None, metas=None, domain=None, text_features=None):
         self.text_features = None    # list of text features for mining
         self._tokens = None
         self._dictionary = None
+        self._ngrams_dictionary = None
+        self._ngrams_matrix = None
```

Comment:
Would be maybe better to store gensim's corpus directly instead of `ngrams_matrix`? Currently, if I get this correctly, in `bagofwords.py` we transforms corpus2csc to get X, in `base.py` we then store X as `ngrams_matrix` and in GensimWrapper we use `Sparse2Corpus` to reconstruct the corpus. Why all the hassle if we could store gensim's corpus directly and save us a few transformations?

An other — maybe a bit too extreme — idea would be to eliminate this all together in favour of using tokens/ngrams directly? Something like this:
```
gcorpus = [self.dictionary.doc2bow(doc) for doc in corpus.tokens]
```
Though I maybe prefer the above idea of having a gensim corpus property.